lib/iris/tests/unit/cube/test_Cube__aggregated_by.py

```diff
@@ -66,9 +66,7 @@ def setUp(self):
 
         self.mock_agg = mock.Mock(spec=Aggregator)
         self.mock_agg.cell_method = []
-        self.mock_agg.aggregate = mock.Mock(
-            return_value=mock.Mock(dtype="object")
-        )
+        self.mock_agg.aggregate = mock.Mock(return_value=np.arange(4))
```

Comment:
Now that we use `stack` instead of assigning to array indices, the mocked aggregate method needs to return an array of the expected shape.